src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Publisher.java

```diff
@@ -30,10 +33,12 @@
 
   private Collator collator;
   private boolean isMuted = false;
+  private List<ActionListener> actionListeners;
```

Comment:
should we make this list thread safe ? it looks to me that addActionListener() is called from a different thread.

Reply:
It should only be called during instantiation of the graph, not during regular scheduling.